src/Symfony/Component/Lock/Store/MongoDbStore.php

```diff
@@ -108,21 +112,25 @@ public function __construct(Collection|Client|string $mongo, array $options = []
         $this->initialTtl = $initialTtl;
 
         if ($mongo instanceof Collection) {
-            $this->collection = $mongo;
+            $this->options['database'] = $mongo->getDatabaseName();
+            $this->options['collection'] = $mongo->getCollectionName();
+            $this->manager = $mongo->getManager();
         } elseif ($mongo instanceof Client) {
-            $this->client = $mongo;
+            $this->options['database'] = $mongo->getDatabaseName();
+            $this->manager = $mongo->getManager();
+        } elseif ($mongo instanceof Manager) {
+            $this->manager = $mongo;
         } else {
             $this->uri = $this->skimUri($mongo);
         }
 
-        if (!($mongo instanceof Collection)) {
-            if (null === $this->options['database']) {
-                throw new InvalidArgumentException(sprintf('"%s()" requires the "database" in the URI path or option.', __METHOD__));
-            }
-            if (null === $this->options['collection']) {
-                throw new InvalidArgumentException(sprintf('"%s()" requires the "collection" in the URI querystring or option.', __METHOD__));
-            }
+        if (null === $this->options['database']) {
+            throw new InvalidArgumentException(sprintf('"%s()" requires the "database" in the URI path or option.', __METHOD__));
```

Comment:
Are these extracted from the connection string somehow? I'll note that `collection` and `database` are not valid URI options, so unless they are removed from the connection string there are potential problems down the road as the driver may reject unknown URI options in the future.

Reply:
The database name is extracted from the URI path. The collection name comes from the query string, but it is removed before the URI is given to MongoDB Manager.